### DISCUSSION TITLE
calculate the number of calls of parent and child applications in real time

### DISCUSSION THREAD

**Discussion by wangquanchn** [open-telemetry/opentelemetry-go#2408] [Q&A]
Hi everyone, 
I successfully got the trace data with otel-go lib, 
and the flow path is this: my-go-app -> collector -> kafka -> java consumer(custom business) -> ... .
now I need to calculate the number of calls of parent and child applications in real time,
when a message is received from kafka,
in every span, I get  trace_id,span_id,parent_id, current span's service_name,
I can't get the parent's service_name. sad.

1. Can I add parent's name to each span?
2. If possible Q1, how to achieve it?

**Comment by dmathieu** [comment]
You can't add the parent's name to each span.
However, you could store the curent service name alongside the span ID whenever you receive one.
Then, for a child span, you could use the parent ID to retrieve the parent span's name.

A key/value store would allow you to solve that easily. You probably don't need more than 15 minutes of data retention, since the links you'll need to do should always be around the same time window.

**Reply by wangquanchn** [reply]
I'm glad you answered my question,

It is a good and easy to implement scheme,
I'll try it,

Thank you very much.
